src/platform/intel/cavs/lib/clk.c

```diff
@@ -143,8 +156,10 @@ static int clock_platform_set_cpu_freq(int clock, int freq_idx)
 
 void platform_clock_init(struct sof *sof)
 {
+	atomic_t *sleep_mask = cache_to_uncache(&sleep_core_mask);
```

Comment:
sleep_core_mask is declared only for CONFIG_CAVS_USE_LPRO_IN_WAITI so this piece will not compile for another config. Impl note: pls consider two versions of a short static inline function,  one containing these extra two init lines (in case of cfg var set to 1) and another one empty, instead of adding #if here inside the platform_clock_init().

Reply:
updated